src/yunohost/certificate.py

```diff
@@ -809,6 +817,15 @@ def _domain_is_accessible_through_HTTP(ip, domain):
     return True
 
 
+def _domain_is_resolved_locally(public_ip, domain):
+    try:
+        ip = socket.gethostbyname(domain)
+    except socket.error:
+        return False
+
+    return (ip in ["127.0.0.1", public_ip])
```

Comment:
Agreed we should ideally support IPv6, but that's gonna take a few more days since it's not trivial from [the documentation of `gethostbyname()`](https://docs.python.org/2/library/socket.html#socket.gethostbyname) ... We already know that the whole cert management (and probably Yunohost) does not support ipv6-only hosts, so ? :/